configure.ac

```diff
@@ -378,6 +378,7 @@ if test "x$with_iconv" != "xno"; then
     AC_CHECK_HEADERS([localcharset.h])
     am_save_LIBS="$LIBS"
     LIBS="${LIBS} ${LIBICONV}"
+    LIBSREQUIRED="$LIBSREQUIRED${LIBSREQUIRED:+ }iconv"
```

Comment:
This is totally broken and breaks all attempted use of the pkg-config file.

Reply:
Yes this is totally broken, e.g. with my OBS build (https://build.opensuse.org/package/binary/home:alvistack/libarchive-libarchive-3.6.2/CentOS_8_Stream/x86_64/libarchive-devel-3.6.2-2.1.x86_64.rpm), nothing could provide `pkgconfig(iconv)`:

![image](https://user-images.githubusercontent.com/780562/206823902-b16ca5b3-e2aa-4ce1-83e9-7993b2516bc5.png)

Manually rollback this patch could help, e.g.:
```
curl -skL https://github.com/libarchive/libarchive/pull/1723.patch | patch -p1 -r
```

Reply:
My apology, I did not checked it in other systems. I have only tried to fix the issue in msys2/mingw environment, see the linked issue above. The msys2/mingw-packages repository provides an iconv.pc file https://github.com/msys2/MINGW-packages/tree/master/mingw-w64-libiconv. Feel free to revert this commit if needed.